frontend/taipy-gui/src/components/Taipy/Metric.spec.tsx

```diff
@@ -12,20 +12,321 @@
  */
 
 import React from "react";
-import {render} from "@testing-library/react";
+import { render, waitFor } from "@testing-library/react";
 import "@testing-library/jest-dom";
+import { ThemeProvider } from "@mui/material/styles";
+import { createTheme } from "@mui/material/styles";
 
 import Metric from "./Metric";
 
+const template = {
```

Comment:
Honestly, for testing purposes, I would have made those far lighter.
But well, go for it.